lib/ecto/query/builder.ex

```diff
@@ -266,6 +266,12 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  def escape({:row_number, _, [arg]}, type, params_acc, vars, env) do
```

Comment:
If row_number is a regular "function", you don't need this here. :)

Reply:
I.e. you can just remove this function and the default transformations will apply.

Reply:
If I just remove this `escape` function, we will get "is not a valid query expression."
`row_number` is a kind of aggregation function, but it can be invoked only over a window, so we can't just mix it with regular aggregators. And there are a bunch of functions like this (`rank`, `dense_rank`, `first_value`, etc) which also have to be implemented. Should I add a `call_type` instead of `escape` for them?

btw, do you have any plans to extend the list of aggregate functions?

Reply:
> Should I add a call_type instead of escape for them?

You are right. `call_type` is the way to go here.

> btw, do you have any plans to extend the list of aggregate functions?

@Anber we could add more functions but it would be nice to know how they are handled across databases. Which ones are supported by all databases (PG, MySQL and MSSQL)? Which ones are PG specific?

Reply:
> Which ones are supported by all databases

Maybe it can be made extendable? DB driver (or even user's code) can provide supported functions to the builder.

Reply:
The thing is that we need the type information to be able to use in the query syntax correctly. So I think for now we should support what is common. Folks can use fragments for everything else.